interpreter/cling/lib/Interpreter/IncrementalParser.h

```diff
@@ -68,9 +71,6 @@ namespace cling {
     // file ID of the memory buffer
     clang::FileID m_VirtualFileID;
 
-    // CI owns it
```

Comment:
I just tested it in master and it does call the dtor :) See this printf output:
```
teemperor@ftlserver ~/r/trunk-build> ninja -j3 Core
[2/97] Generating G__Core.cxx, ../../lib/libCore.rootmap
DEBUG: DESTROYED DeclCollector
[97/97] Linking CXX shared library lib/libCore.so
```
You probably tested rootcling, but rootcling_stage1 has a different memory ownership than rootcling and does destroy the DeclCollector. That's why Core is crashing.